### PR TITLE
Cleanup the pipeline code and use PipelineResourceCollection so main is cleaner

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

File: gnomad_constraint/pipeline/constraint_pipeline.py [hunk 1]
```diff
@@ -456,17 +488,22 @@ def main(args):
...
+    prepare_context_args = parser.add_argument_group(
+        "Prepare context Table args", "Arguments used for preparing the context Table."
     )
-    preprocess_data_args.add_argument(
+    prepare_context_args.add_argument(
         "--prepare-context-ht",
         help=(
-            "Whether to split multiallelic sites and add 'methylation', 'coverage', and"
-            " 'gerp' annotation to context Table with VEP annotation."
+            "Prepare the context Table by splitting multiallelic sites and adding "
```
Comment (klaricch) [inline gnomad_constraint/pipeline/constraint_pipeline.py:497]: aren't multiallelic sites already split?

File: gnomad_constraint/pipeline/constraint_pipeline.py [hunk 2]
```diff
@@ -456,17 +488,22 @@ def main(args):
...
+            "'methylation', 'coverage', and 'gerp' annotation to context Table with "
```
Comment (klaricch) [inline gnomad_constraint/pipeline/constraint_pipeline.py:498]:
```suggestion
            "'methylation', 'coverage', and 'gerp' annotations to context Table with "
```

**Review by klaricch — CHANGES_REQUESTED** [review]

File: gnomad_constraint/pipeline/constraint_pipeline.py (hunk 1 above)
Comment (jkgoodrich) [reply gnomad_constraint/pipeline/constraint_pipeline.py:497]: yeah, I was confused about that too, maybe the original one used was not. I can either remove that or add a check to see if it's split and only split if not. Thoughts?

Comment (klaricch) [reply gnomad_constraint/pipeline/constraint_pipeline.py:497]: i like adding a check and only splitting if necessary

File: gnomad_constraint/pipeline/constraint_pipeline.py [hunk 3]
```diff
@@ -72,135 +54,236 @@
...
+        "--prepare-context-ht",
+        output_resources={
+            "annotated_context_ht": constraint_res.get_annotated_context_ht(
+                version, use_v2_release_context_ht
+            )
+        },
+        input_resources={
+            "gnomAD resources": {
+                "context_ht": context_res,
+                "exomes_coverage_ht": constraint_res.get_coverage_ht("exomes"),
```
[the remark this answers is not in the document]
Comment (klaricch) [reply gnomad_constraint/pipeline/constraint_pipeline.py:121]: nm, i had imported an old version of check_param_scope which messed it up

**Review by klaricch — APPROVED** [review]
lgtm

**Pull request merged**